compiler-rt/lib/mpk_untrusted/mpk_fault_handler.cpp

```diff
@@ -90,7 +90,10 @@ void disableMPK(siginfo_t *si, void *arg) {
 void stepMPKHandle(int sig, siginfo_t *si, void *arg) {
   __sanitizer::Report(
       "Reached signal handler after single instruction step.\n");
-  enableThreadMPK(arg, si->si_pkey);
+  auto handler = __mpk_untrusted::AllocSiteHandler::get();
+  auto pid_key_info = handler->getPidKey(getpid());
```

Comment:
does this get a handler for a specific thread? also, doesn't `getpid()` return the process id and not the thread_id? i'm not sure what this does, given that `getpid()` should return the same integer for all threads. `sys/types.h` and `unistd.h` together define `gettid()` that will give you the thread id. only in single threaded applications will `getpid() == gettid()`.

Reply:
This call will get a shared reference to a singleton handler, then each access function ensures correct access patterns by using the mutex within the handler. Also switched to `gettid()` macro in accordance with suggestion, and added tests to ensure the macro functioned correctly (and as expected).